app/lib/core/utilities/dialog_manager.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:app/core/models/dialog.dart';
+import 'package:app/core/services/dialog/dialog_service.dart';
+import 'package:app/core/services/service_locator.dart';
+import 'package:flutter/material.dart';
+
+class DialogManager extends StatefulWidget {
+  final Widget child;
+  DialogManager({Key key, this.child}) : super(key: key);
+
+  _DialogManagerState createState() => _DialogManagerState();
+}
+
+class _DialogManagerState extends State<DialogManager> {
+  DialogService _dialogService = locator<DialogService>();
+
+  @override
+  void initState() {
+    super.initState();
+    _dialogService.registerDialogListener(_showDialog);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return widget.child;
+  }
+
+  void _showDialog(DialogRequest request) {
+    var isConfirmationDialog = request.cancelText != null;
```

Comment:
Would prefer optional `cancelText` in `DialogRequest` than check if null.

Reply:
Will do, thanks!